cognee/infrastructure/databases/vector/neptune_analytics/NeptuneAnalyticsAdapter.py

```diff
@@ -206,6 +218,14 @@ async def search(
                 "Use this option only when vector data is required."
             )
 
+        # In the case of excessive limit, or zero / negative value, limit will be set to 10.
+        if not limit or limit < 0 or limit > 10:
```

Comment:
```suggestion
        if not limit or limit <= 0 or limit > 10:
```
Do we want to give a maximum?

Reply:
We should make this a constant in the class, to make it easier to test/maintain.